url5583f3dbb2_XOR19_XScript_tgz-pJ8-xscript_interactive_MainJ8/src/xscript/XExec.java

```diff
@@ -0,0 +1,1927 @@
+			break;
+		}
+		case CALL_MAP_KW: {
+			i1 = callFrame.readUByte() - 1;
+			int i2 = callFrame.readUByte();
+			v3 = null;
+			XValue[] params = new XValue[i1];
+			for (int i = i1, j = i1 - 1; i >= 0; i--, j--) {
+				if (i == i2) {
+					v3 = pop();
+					i--;
+					continue;
+				}
+				params[j] = pop();
+			}
+			String[] kws = new String[i1];
+			int i3 = callFrame.readUByte();
+			for (int i = 0; i <= i1 - i3; i++) {
+				kws[i] = callFrame.readStringP();
+			}
+			v2 = pop();
+			v1 = pop();
+			call(v2, v1, params, kws, null, v3);
+			break;
+		}
+		case COMPARE:
+			v1 = pop();
+			v2 = pop();
+			if (v1.isNumber() && v2.isNumber()) {
+				if (v1.isFloat() || v2.isFloat()) {
+					i1 = Double.compare(v2.getFloat(), v1.getFloat());
+				} else {
+					i1 = Long.compare(v2.getInt(), v1.getInt());
+				}
+				v3 = XValueInt.valueOf(i1);
+				push(v3);
+			} else {
+				callBiOperator("<=>", "__compare__", "__compare2__", v2, v1);
+			}
+			break;
+		case COPY:
+			v1 = pop();
+			if (v1.isObject()) {
+				callSingleOperator("<:", "__copy__", v1);
+			} else {
+				push(v1);
+			}
+			break;
+		case DEC:
+			v1 = pop();
+			if (v1.isNumber()) {
+				if (v1.isFloat()) {
+					v2 = new XValueFloat(v1.getFloat() - 1);
+				} else {
+					v2 = XValueInt.valueOf(v1.getInt() - 1);
+				}
+				push(v2);
+			} else {
+				callSingleOperator("--", "__dec__", v1);
+			}
+			break;
+		case DEL_ATTR:
+			s = callFrame.readStringP();
+			v1 = pop();
+			if (handles__getItem__(v1)) {
+				v2 = XUtils.lookup(rt, v1, "__delItem__", XValue.REF_NONE);
+				v3 = rt.alloc(s);
+				call(v2, v1, new XArrayList<XValue>(v3), null);
+			} else {
+				XUtils.del(rt, v1, s, XValue.REF_NONE);
+			}
+			break;
+		case DEL_GLOBAL:
+			s = callFrame.readStringP();
+			v1 = callFrame.getModule();
+			XUtils.del(rt, v1, s, XValue.REF_NONE);
+			break;
+		case DEL_INDEX:
+			v1 = pop();
+			v2 = pop();
+			v3 = XUtils.lookup(rt, v2, "__delIndex__", XValue.REF_NONE);
+			call(v3, v2, new XArrayList<XValue>(v1), null);
+			break;
+		case DIV:
+			v1 = pop();
+			v2 = pop();
+			if (v1.isNumber() && v2.isNumber()) {
+				v3 = new XValueFloat(v2.getFloat() / v1.getFloat());
+				push(v3);
+			} else {
+				callBiOperator("/", "__div__", "__div2__", v2, v1);
+			}
+			break;
+		case DUP:
+			v1 = getTop(0);
+			push(v1);
+			break;
+		case END_FINALLY:
+			v1 = pop();
+			v2 = pop();
+			i1 = (int) v1.getInt();
+			if (i1 == 0) {
+				_throw(v2);
+			} else if (i1 == 1) {
+				List<XValue> list = XUtils.asList(rt, v2);
+				if (list != null) {
+					if (list.size() == 1) {
+						XValue nextJump = list.get(0);
+						int jump = (int) nextJump.getInt();
+						callFrame.jumpTo(jump);
+					} else {
+						list = new ArrayList<XValue>(list);
+						XValue nextJump = list.remove(list.size() - 1);
+						v3 = rt.createTuple(list);
+						push(v3);
+						push(v1);
+						int jump = (int) nextJump.getInt();
+						callFrame.jumpTo(jump);
+					}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 811:819, 826:834.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.